RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,174 @@
+
+The HTLC is constructed by locking an output with the following script:
+
+```asm
+OP_IF
+    OP_SIZE 32 OP_EQUALVERIFY
+    OP_SHA256 <secret_hash> OP_EQUALVERIFY
+    OP_DUP OP_HASH160 <redeem_identity>
+OP_ELSE
+    <timestamp> OP_CHECKLOCKTIMEVERIFY OP_DROP
+    OP_DUP OP_HASH160 <refund_identity>
+OP_ENDIF
+OP_EQUALVERIFY
+OP_CHECKSIG
+```
+
+As required by RFC003, this HTLC uses absolute time locks to check whether `expiry` has been reached.
+Specifically, `OP_CHECKLOCKTIMEVERIFY` (see [BIP65](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki)) is used to compare the time in the block's header to the `expiry` in the contract.
+
+To compute the exact bytes of the contract implementaions should use the following offset table:
+
+| Data              | Position of first byte | Position of last byte | Length | Description                                                                |
+|:------------------|:-----------------------|:----------------------|:-------|:---------------------------------------------------------------------------|
+| `6382012088a820`  | 0                      | 6                     | 7      | `OP_IF OP_SIZE 32 OP_EQUALVERIFY OP_SHA256` + `PUSH32` for the secret_hash |
+| `secret_hash`     | 7                      | 39                    | 32     | See [Parameters](#parameters)                                              |
+| `8876a9`          | 40                     | 42                    | 3      | `OP_EQUALVERIFY OP_DUP OP_HASH160`                                         |
+| `redeem_identity` | 43                     | 74                    | 32     | See [Parameters](#parameters)                                              |
+| `67`              | 75                     | 75                    | 1      | `OP_ELSE`                                                                  |
+| `timestamp`       | 76                     | 79                    | 4      | See [Parameters](#parameters)                                              |
+| `b17576a9`        | 80                     | 83                    | 4      | `OP_CHECKLOCKTIMEVERIFY OP_DROP OP_DUP OP_HASH160`                         |
+| `refund_identity` | 84                     | 115                   | 32     | See [Parameters](#parameters)                                              |
+| `6888ac`          | 116                    | 118                   | 3      | `OP_ENDIF OP_EQUALVERIFY OP_CHECKSIG`                                      |
+
+
+## Execution Phase
+
+### Deployment
+
+The party who is required to deploy the Bitcoin HTLC (the refunder) compiles the contract as described in the previous section.
+Then, they MUST send a transaction to the Bitcoin Blockchain with an output with value exactly equal to the quantity of satoshi specified by `asset` and a Pay-To-Witness-Script-Hash (P2WSH) `scriptPubKey` derived from the contract. See [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#specification)
```

Comment:
:question: Is this really a `MUST`? You don't use this sort of language in the rest of this section anyway, but to me this is a `MAY` unless you want to have an opinion on whether the "funder"/"deployer"/"refunder" should proceed with the protocol. What if Bitcoin is `beta_ledger` and Alice took too long to fund `alpha_asset`?

Reply:
I changed it :)